### PR TITLE
Prune sample sets if a sample query is given

### LINKED ISSUE #225: Possible small optimisation - prune sample sets using sample query
A common pattern is to not specify a value for the `sample_sets` parameters, which defaults to all available sample sets, then give a `sample_query`. This is convenient, but then can cause a few delays because not all sample sets are needed when accessing other data like SNP calls.

A possible small optimisation would be to use the sample query to prune the list of sample sets, before doing anything else.

### PR DESCRIPTION
**Pull request by alimanfoo**
* Resolves #225

### CODE REVIEW

**Comment by alimanfoo** [comment]
Hm, how to avoid sample sets progress being shown twice?

File: malariagen_data/ag3.py [hunk 1]
```diff
@@ -1004,7 +1018,9 @@ def snp_genotypes(
         debug = self._log.debug
 
         debug("normalise parameters")
-        sample_sets = self._prep_sample_sets_arg(sample_sets=sample_sets)
+        sample_sets = self._prep_sample_sets_arg(
+            sample_sets=sample_sets, sample_query=sample_query
```
Comment (cclarkson) [inline malariagen_data/ag3.py:1022]: I'm not sure where `sample_query` is coming from here, it doesn't appear a parameter in `def species_calls()`

File: malariagen_data/ag3.py [hunk 2]
```diff
@@ -638,6 +643,15 @@ def _prep_sample_sets_arg(self, *, sample_sets):
                     f"Bad value for sample_sets parameter, {k:!r} selected more than once."
```
Comment (cclarkson) [inline malariagen_data/ag3.py:643]: could we just take unique sample sets here?

Comment (cclarkson) [inline malariagen_data/ag3.py:643]: could we just take unique sample sets here?

**Comment by alimanfoo** [comment]
Probably should benchmark to confirm this does save time.

Also sample metadata gets loaded twice, so two progress bars get shown, which is not so great.

**Pull request closed without merging**